lib/octopus_serverspec_extensions/type/octopus_deploy_account.rb

```diff
@@ -0,0 +1,126 @@
+require 'serverspec'
+require 'serverspec/type/base'
+require 'net/http'
+require 'json'
+
+module Serverspec::Type
+  class OctopusDeployAccount < Base
+    @account = nil
+    @serverUrl = nil
+    @apiKey = nil
+    @serverSupportsSpaces = nil
+    @spaceId = nil
+    @spaceFragment = ""
+
+    def initialize(serverUrl, apiKey, account_name, spaceID = 'Spaces-1')
+      @name = "Octopus Deploy Account #{account_name}"
+      @runner = Specinfra::Runner
+      @serverUrl = serverUrl
+      @apiKey = apiKey
+      @spaceId = spaceID
+
+      if serverUrl.nil?
+        raise "'serverUrl' was not provided. Unable to connect to Octopus server to validate configuration."
+      end
+      if apiKey.nil?
+        raise "'apiKey' was not provided. Unable to connect to Octopus server to validate configuration."
+      end
+      if account_name.nil?
+        raise "'account_name' was not provided. Unable to connect to Octopus server to validate configuration."
+      end
+
+      @serverSupportsSpaces = check_supports_spaces(serverUrl)
+
+      if @serverSupportsSpaces
+        @spaceFragment = "#{@spaceId}/"
+      end
+
+      @account = get_account_via_api(serverUrl, apiKey, account_name)
+    end
+
+    def exists?
+      (!@account.nil?) && (@account != [])
+    end
+
+    def has_description?(account_description)
+      return false if @account.nil?
+      @account["Description"] == account_description
+    end
+
+    def is_account_type?(account_type_name)
```

Comment:
I wonder if some kind of inheritance is the right structure here.
Really depends what the end goal is...